File: 1/1000s/18s/1833/MaximumIceCreamBars.java
/*
https://leetcode.com/problems/maximum-ice-cream-bars/?envType=daily-question&envId=2026-06-21
*/

class Solution {
    public int maxIceCream(int[] costs, int coins) {
        Arrays.sort(costs);
        int lengthOfCosts=costs.length, resultantMaxIceCreams=0, beginIndex=0;
        while(true){
            if(coins==0||beginIndex>=lengthOfCosts) break;
            if(coins-costs[beginIndex]>=0){
                resultantMaxIceCreams++;
                coins-=costs[beginIndex];
            }
            beginIndex++;
        }
        return resultantMaxIceCreams;
    }
}

File: 1/1000s/18s/1833/MaximumIceCreamBars.kt
/*
https://leetcode.com/problems/maximum-ice-cream-bars/?envType=daily-question&envId=2026-06-21
*/

class Solution {
    fun maxIceCream(costs: IntArray, coins: Int): Int {
        costs.sort()
        var coinsTracking:Int=coins
        val lengthOfCosts:Int=costs.size
        var resultantMaxIceCreams:Int=0
        var beginIndex:Int=0
        while(true){
            if(coinsTracking==0||beginIndex>=lengthOfCosts) break
            if(coinsTracking-costs[beginIndex]>=0){
                resultantMaxIceCreams++
                coinsTracking-=costs[beginIndex]
            }
            beginIndex++
        }
        return resultantMaxIceCreams
    }
}